cmd/neofs-cli/modules/container/list_objects.go

```diff
@@ -44,40 +48,52 @@ var listContainerObjectsCmd = &cobra.Command{
 		if err != nil {
 			return err
 		}
+		bt, err := common.ReadBearerToken(cmd, objectCli.BearerTokenFlag)
+		if err != nil {
+			return err
+		}
 		cli, err := internalclient.GetSDKClientByFlag(ctx, commonflags.RPC)
 		if err != nil {
 			return err
 		}
 
-		var prmSearch internalclient.SearchObjectsPrm
 		var prmHead internalclient.HeadObjectPrm
-
-		prmSearch.SetClient(cli)
-
 		if flagVarListObjectsPrintAttr {
 			prmHead.SetClient(cli)
 			prmHead.SetPrivateKey(*pk)
-			err = objectCli.Prepare(cmd, &prmSearch, &prmHead)
-			if err != nil {
-				return err
-			}
-		} else {
-			err = objectCli.Prepare(cmd, &prmSearch)
+			err = objectCli.Prepare(cmd, &prmHead)
 			if err != nil {
 				return err
 			}
 		}
 
-		prmSearch.SetPrivateKey(*pk)
-		prmSearch.SetContainerID(id)
-		prmSearch.SetFilters(*filters)
-
-		res, err := internalclient.SearchObjects(ctx, prmSearch)
-		if err != nil {
-			return fmt.Errorf("rpc error: %w", err)
+		var opts client.SearchObjectsOptions
+		ttl := viper.GetUint32(commonflags.TTL)
+		common.PrintVerbose(cmd, "TTL: %d", ttl)
+		if ttl == 1 {
+			opts.DisableForwarding()
 		}
+		if bt != nil {
+			opts.WithBearerToken(*bt)
+		}
+		opts.WithXHeaders(objectCli.ParseXHeaders(cmd)...)
 
-		objectIDs := res.IDList()
+		var objectIDs []oid.ID
+		var res []client.SearchResultItem
+		var cursor string
+		for {
+			res, cursor, err = cli.SearchObjects(ctx, id, *filters, nil, cursor, (*neofsecdsa.Signer)(pk), opts)
+			if err != nil {
+				return fmt.Errorf("rpc error: %w", err)
+			}
+			objectIDs = slices.Grow(objectIDs, len(objectIDs)+len(res))
+			for i := range res {
+				objectIDs = append(objectIDs, res[i].ID)
```

Comment:
Can be huge, we better process (output) them chunk by chunk. It'd also be more user-friendly, no big delay before the first result seen.

Reply:
thought about this. This is a behavior change (we could do this w/ old search too, but did not), but useful, so will add above